### ISSUE TITLE
crash in selfplay

### ISSUE THREAD

**Issue by iopq** [lightvector/KataGo#237]
```
terminate called after throwing an instance of 'StringError'
  what():  CUDA Error, for getOutput file /home/iopq/code/kata/cpp/neuralnet/cudabackend.cpp, func cudaMemcpy(inputBuffers->valueResults, buffers->valueBuf, inputBuffers->singleValueResultBytes*batchSize, cudaMemcpyDeviceToHost), line 3016, error unspecified launch failure
./blockingloop.sh: line 50: 10706 Aborted                 (core dumped) "$KATAEXEC" selfplay -output-dir "$BASEDIR"/selfplay -models-dir "$BASEDIR"/models -config-file selfplay1.cfg
```

ubuntu 19, cuda 10.1 (AFAIK, I could check)

what could it be caused by? I recently woke up the computer and resumed the process

**Comment by lightvector** [comment]
Not sure. Could be some sort of intermittent GPU failure, or overheating, or something. Or KataGo has some super-rare memory corruption bug or something.

Intermittent GPU failure is actually a thing - I've had one of my KataGo processes on the cloud fail for no good reason on some CUDA calls before that normally never fail, and in that particular case, around the same time I got an AWS notification that instability had been detected with the GPU and that I needed to shut down and restart the instance (upon which it would start with different GPU, the old GPU presumably to be taken offline for maintenance).
